astropy/table/column.py

```diff
@@ -711,6 +711,13 @@ def __new__(cls, data=None, name=None,
                                           copy=copy, copy_indices=copy_indices)
         return self
 
+    def __set_attr__(self, item):
```

Comment:
You mean `__setattr__`.

That said, I'm not sure overriding `__setattr__` is the best approach to this, since it will slow down all attribute access (granted, by a only a small amount).

Instead I think the `Column` class could get a `.mask` property defined to raise an exception:

```
@property
def mask(self):
    raise AttributeError('mask')
```

Should see what @taldcroft thinks though.